psi4/src/psi4/dfocc/manager.cc

```diff
@@ -3485,23 +3485,25 @@ void DFOCC::lccd_manager() {
     outfile->Printf("\tDF-MP2 Total Energy (a.u.)         : %20.14f\n", Emp2);
     outfile->Printf("\t======================================================================= \n");
 
-    Process::environment.globals["MP2 TOTAL ENERGY"] = Emp2;
+    variables_["MP2 TOTAL ENERGY"] = Emp2;
     Process::environment.globals["SCS-MP2 TOTAL ENERGY"] = Escsmp2;
     Process::environment.globals["SOS-MP2 TOTAL ENERGY"] = Esosmp2;
     Process::environment.globals["SCS(N)-MP2 TOTAL ENERGY"] = Escsnmp2;
-    Process::environment.globals["MP2 CORRELATION ENERGY"] = Emp2 - Escf;
+
+    variables_["MP2 CORRELATION ENERGY"] = Emp2 - Escf;
```

Comment:
What's the reasoning behind this change? I don't quite get why we're shifting some variables into `variables_` out of the global namespace, but not others. Is it to attach them to wfn?

Reply:
Exactly on the attaching qcvars to wfn. `variables_` sets on wfn, then in proc.py, wfn vars get copied into `P::e`. That's for the variable names I can vouch for. My current focus is totals and components. SCS variations are another pass, and I may kill off some of them so that `Wfn.variables()` is a fresh start from `core.variables()`. Someday `P::e.globals` will be only a pseudo-global that gets written to at the end of `energy()`, etc. for user convenience.

Reply:
That makes sense. Thanks.